Sources/GRPC/ClientCalls/BaseClientCall.swift

```diff
@@ -99,6 +105,7 @@ open class BaseClientCall<RequestMessage: Message, ResponseMessage: Message> {
     self.status = self.responseHandler.statusPromise.futureResult
 
     self.streamPromise.futureResult.whenFailure { error in
+      self.logger.error("failed to create http/2 stream: \(error)")
```

Comment:
Should the `error` parameter be part of the metadata? (I'm unfamiliar with best practices here.)

Reply:
Good question, I'm not super familiar with best practices here either! 

From the docs:
> Logging metadata is metadata that can be attached to loggers to add information that is crucial when debugging a problem

It seems like it could be either: I suppose adding it to metadata would make it easier for handlers to extract the description of the actual `Error` the message is related to which could be useful for e.g. emitting metrics.

@weissi do you have any insight here on when to when use metadata vs. include it in the log message? Things like identity (request Id, connection Id) seem obvious since they're useful for separating out instances. Things like errors are less clear.